### ISSUE TITLE
Adding Peercoin support to Keybase for distributed data application PeerApps

### ISSUE THREAD

**Issue by emeth-** [keybase/keybase-issues#1697]
Hey there,

I am the author of PeerApps, and its child projects such as PeerMessage.

https://github.com/Peerapps/Peerapps/tree/master/peermessage

This is a python-based application that runs alongside a peercoin wallet. It adds a layer over the protocol that adds support for distributed data applications - the "key" is stored on the blockchain in an op_return transaction and points to the value, which is currently stored in various parasitic data stores (tinyurls, pastebin, etc). 

One of the main initial applications right now is PeerMessage, linked above, which lets the user create a GPG keyset and publish their public key on the blockchain, then enables communication back and forth.

The next project is PeerMarkets, which will be an ebay-like competitor that utilizes PeerMessage for communication. In an effort to avoid creating another Silk Road, the PeerApps system requires people to establish an identity (a single Peercoin address) on which their reputation is based.

Keybase fits perfectly into this project. I could require users to setup a keybase account and manage their GPG keys and identities there, and just focus PeerApps on the application layer between Keybase and the Peercoin blockchain. 

If you guys could add Peercoin support similarly to how you added Bitcoin, I would happily start on integrating it immediately. For the purposes of my app, my users will establish keybase.io accounts, and my app will mainly just need the ability to submit a peercoin address to your api and retrieve the associated user's GPG public key and social account info.

I would also love an invite - I'm on the waiting list, not sure how long that takes.

**Comment by mh001** [comment]
Yes. Peercoin is the sutainable and durable substitution for bitcoin.

**Comment by bananenwilly** [comment]
Peercoin would go along nicely with keybase. The sinergy effects look very promising.

**Comment by Sentinelrv** [comment]
Hi, addition of Peercoin would be very much appreciated by our community as we won't need to create our own solution anymore and could instead use Keybase. In case you don't know, Peercoin was the original crypto to introduce proof-of-stake consensus technology (compared with Bitcoin's proof-of-work) and is the first energy efficient crypto. Peercoin was also designed by Sunny King as a backbone currency which prioritizes decentralization and secure store of value over being a pure transactional currency. The addition of Peercoin support with Keybase would really help out our upcoming projects based on the Peerapps technology that Emeth has developed.

**Comment by mwoliver** [comment]
Indeed, the addition of Peercoin to keybase would be most welcome by many.

**Comment by Marak** [comment]
+1

**Comment by ppcpal** [comment]
Adding Peercoin support to keybase will be very exciting.

**Comment by Captain-Harlock** [comment]
We definitely need Peercoin support.

**Comment by rootsical** [comment]
i would also like to see peercoin support added..

**Comment by Lamz0rNewb** [comment]
I can only try to imagine how greatly keybase and PeerApps might profit from each other!
Peercoin has a very innovative and dedicated community (which can be seen at projects like PeerApps!). Let's bring decentralization and security forward through joint efforts!
Please consider Peercoin to keybase.

**Comment by Sebsebzen** [comment]
Please add Peercoin support!

**Comment by FuzzyBearBTC** [comment]
Keen to see Peercoin support added to keybase if you want to raise fund for development please feel free to list the github repo on http://peer4commit.com/

**Comment by Sentinelrv** [comment]
Yes, Fuzzy makes a good point here. We use peer4commit.com as a way to raise funding for projects. If you would like to be paid for this implementation, just open up a project on that website and list how much we should raise. We'll then do our best to raise the necessary funds for the work you need to do.

**Comment by jointhepartypooper** [comment]
:+1:

**Comment by dtiersch** [comment]
@emeth- 
have you already received an invitation? I have some left and could invite you. But I need your email address for this (encrypt [here](https://keybase.io/encrypt#salandas))

**Comment by emeth-** [comment]
Not yet - thanks dtiersch!

-----BEGIN PGP MESSAGE-----
Version: Keybase OpenPGP v2.0.20
Comment: https://keybase.io/crypto

wcNMA75WDYsDMCUYASAAjz4nF2GAonYpx0zrScWpqh5Qjr2bLAuCe//yMLIdZ7Ym
CPZVXGGlvoG4kgep0cwVqbWUPsz7FykUFEg+estBzAQkqkmloffENT62mpznYbn1
/5J5ugHY/HUa7CZY+VCJBfywn2eSTxBhMG37Uq64zr4ZVzLa9lW34ZQwQva9k0cJ
t3pMb2vHRJjlqHMu/WJvERUOqzYWrWcJ9F18niIBqQWPDLPnZBD6kRnUQ7NHDgIi
PZ4u31Xt/h8gzaeC7p4iWOgtRrD7PfLCbmLwiUoXl2D4qH1UkFBKE4qDibts+KfN
pIX7RRsx0I7OWFXl3xIHTDrF+vL1tiuvyO+TXMzsWiodzgz7pbgg2MG2C8QPnSNf
JN9bXjPlg3eFo2duv3+VYCZphH4qw++uR4jLz4P/s/44JsUMp01KzonKe72PhUrH
xNsZWqmzHxb5GkfezwIhHk7JWNJNkWrBjdz6nvH4vh7ibxmbvyR+1X4pYg0DosKZ
1Ka36/qnZ++WuDt3PSDVGrmN9Yl7x+O4/2wRtew2bsLIiZC6MzguH7HecCHay+gC
KWrvp/VmTSHhgXfq4AOX3pQDVCbsqKi4D/25vDn9McPwXwZqmhNBMcpKNFeyBYJZ
6p6UN4Had757HkD1H5RYNtaIn+lyfRUrV129MsWt2NXvcbWgxRCaIDVJMCzLeYYq
SWsuv52DIqLOd/tlPXDuqNlKvknzOcyyxX5/tK28gecKG0iNd2PMSjdNaLmG44n9
kfxZkSs0Wct91x8FdsoN3AuuzAJtpmZuIUDaLn/tExORfpV5mLJWRFekq+b1SAix
L18l45G6opasYxog+uCoqYg12dVMqt6wV9Kyu50Tmt4QGOe8Y0aqNXMofWAjpQrG
kPfLiBqdk88L6TnFJJHCoWfW848Tq6Nt7CvpeH3iZJxBhDWiXT622MovoQfAKZv5
W814GHKP2bsF53C2hiWa7SJEnwP6XHC28DobUlYNVOi9yftReDpIAJEqkD5SPX9c
MlFuGxXT9STP0xrzs3mA16eUezULnu13/XKDHAIVBZDLhPyIKdKOhnD0i0ynixpJ
dI9+6hQDcppk/m73hyPJ+YtwO9D0af0/dY+wQQaPi7ameGZ3b4ajm5SnvGYoup8y
lXYNDJlg8NGHWP6/fCFrv/tdKiOMDe0ojEPiKFjACL9Ctw7tWBxTv0Fa1lfvcL2i
edgbcqWw3wUTPsudbX6GXr1dx5L4KWaihZYhL7AisLU14HcRCdOUJfdLCmPYMr28
snhdE5OyDwafl3+PuyH91w982pinOIDEgJ/HObYzCplBSvsFll4j4ouR1wSTd4F8
HXa2Gqjkto+4vlDzI/kk9UFOCRXHAhEcdaa5+7E6Z9JSAbAPC/mSZTwu/UNxw6ra
lk7X5ChPYOU0XD4Mis7OvXayRul6JBz+Od6m0gKpXuV1yYwZTeBkx0Pb//wHvNKo
KTAkdf9x1ArgVuZAfJvFXs/crw==
=hKmW
-----END PGP MESSAGE-----

**Comment by Cybnate** [comment]
Bit late to the party, but still like to support this.

**Comment by jooize** [comment]
I initially didn't understand how these would work together, but after some thinking it sounds like a good idea. I support this as well.

**Comment by weeeee** [comment]
Please add Peercoin support

Thank you

**Comment by embeddedthought** [comment]
PeerApps and Keybase can be coupled together to develop an exceedingly expansive range of applications. Please take the time to consider integrating support for Peercoin. We're anxious to get started.

**Comment by jooize** [comment]
Could this be used to build a [single sign-on](https://en.wikipedia.org/wiki/Single_sign-on) solution and replace [OpenID](https://en.wikipedia.org/wiki/OpenID)/[OAuth](https://en.wikipedia.org/wiki/OAuth) systems?

**Comment by adsactly** [comment]
Not sure why nobody pursued this at the Keybase Development Team.

**Comment by skwerlman** [comment]
@adsactly If you look around this issue tracker, you'll see that there are _literally hundreds_ of similar requests. Keybase devs only have so much time on their hands (they are, surprisingly, mortal), so they need to prioritize what they work on. Currently, it seems like their priority is making sure the client and web interfaces are as nailed down and bug-free as possible, before they move onto mobile apps, and then the infinite slog of user requests for site integrations.

Please try to understand that this is still a relatively young project run by real human people who have to solve some very hard problems before they can get started on increasing the breadth of their integrations.
